lib/services/lookup.js

```diff
@@ -317,27 +331,28 @@ function lookupServiceFactory(
      */
     LookupService.prototype.ipAddressToNodeId = function (ip) {
         assert.isIP(ip);
-
-        var cachePromise = this.checkNodeIdCache(ip);
-
+        
+        var self = this;
+        var cachePromise = self.checkNodeIdCache(ip);
         if (cachePromise) {
             return cachePromise;
         }
-
-        return waterline.lookups.findOneByTerm(ip).then(function (record) {
-            if (record.node) {
-                return this.assignNodeIdCache(ip, record.node);
-            }
-
-            else {
-                this.clearNodeIdCache(ip);
-
-                throw new Errors.NotFoundError(
-                    'Lookup Record Not Found (ipAddressToNodeId)',
-                    record
-                );
-            }
-        }.bind(this)).catch(this.clearNodeIdCacheAndThrow.bind(this, ip));
+        
+        return self.validateArpCache()
+        .then(function() {
+            return waterline.lookups.findOneByTerm(ip).then(function (record) {
+                if (record.node) {
+                    return self.assignNodeIdCache(ip, record.node);
+                }
+                else {
+                    self.clearNodeIdCache(ip);
+                    throw new Errors.NotFoundError(
+                        'Lookup Record Not Found (ipAddressToNodeId)',
+                        record
+                    );
+                }
+            }.bind(self)).catch(self.clearNodeIdCacheAndThrow.bind(self, ip));
```

Comment:
Binding to `self` is redundant if you are using `self` references within the callback. If you wanted to use `this` references, then it would make more sense (as some of the other code appears to be doing). It's inconsequential, but slightly confusing.